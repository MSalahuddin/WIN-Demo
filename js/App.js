/* eslint-disable no-nested-ternary */
import React, { useEffect} from 'react';
import { StatusBar, LogBox } from 'react-native'
import AsyncStorage from '@react-native-community/async-storage';
import { GetSocial } from 'getsocial-react-native-sdk';
import UserProvider, { UserContext } from './context/User.context';


import PopupProvider from './context/Popup.context';
import BackgroundMusicProvider from './context/BackgroundMusic.context';
import Navigation from './components/navigation';
import {
  LOCAL_STORAGE_NAME,
} from './constants';

if (__DEV__) {
  import('./ReactotronConfig');
}

LogBox.ignoreLogs(['Warning: ...']); // Ignore log notification by message
LogBox.ignoreAllLogs(); //Ignore all log notifications


function App() {
  // We don't use Async Storage directly. This message came from Reactotron which would
  // only appear in dev


  const saveReferrerPayLoad = async (payload) => {
    await AsyncStorage.setItem(LOCAL_STORAGE_NAME.REFERRER_DATA, JSON.stringify(payload));
  }

  const getSocialHandler = () => {
    GetSocial.registerForPushNotifications()
    GetSocial.onNotificationReceived((notification, wasClicked) => {
      try {
        const data = JSON.parse(JSON.stringify(notification.action))
        const DATA = JSON.parse(data)
        saveReferrerPayLoad(DATA)
        if (wasClicked) {
          // click silently
        }
      }
      catch (e) {
        // catch silently
      }
    });
  }


  useEffect(() => {
    GetSocial.init()
    getSocialHandler()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <BackgroundMusicProvider>

      <PopupProvider>
        <UserProvider>

          <StatusBar hidden />
          <Navigation />
        </UserProvider>
      </PopupProvider>

    </BackgroundMusicProvider>
  );
}

export default App;
