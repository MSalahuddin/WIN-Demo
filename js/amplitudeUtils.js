import RNAmplitude from 'react-native-amplitude-analytics';
import Config from 'react-native-config';

export const amplitude = new RNAmplitude(Config.AMPLITUDE_API_KEY);
export const logEvent = (eventName, eventProperties = null) => amplitude.logEvent(eventName, eventProperties);
export const setUserId = userId => amplitude.setUserId(userId);
export const setUserProperties = userProperties => amplitude.setUserProperties(userProperties);
export const increaseUserProperty = (property, amount = 1) => amplitude.addToUserProperty(property, amount);
export const logRevenueV2 = (revenueProperties) => amplitude.logRevenueV2(revenueProperties);