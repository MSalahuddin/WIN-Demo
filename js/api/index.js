/* eslint-disable max-len */
/* istanbul ignore file */
// import { Platform } from 'react-native';
import client from './client';
import { WW_API_ENDPOINTS } from '../constants';


  const playerChallengeRank = (challengeId, challengeTypeEnum) => client.get(WW_API_ENDPOINTS.PLAYER_CHALLENGES_RANK(challengeId, challengeTypeEnum));
const getMachines = (params) => client.get(WW_API_ENDPOINTS.GET_GAMEROOM_MACHINES,{ params: { ...params, includeInactive : false } });
const getMachinesPrizes = (machineId, countryId) => 
  client.get(WW_API_ENDPOINTS.GET_GAMEROOM_MACHINES_PRIZES(machineId, countryId));
const getMachinesSocketBaseUrl = () => client.get(WW_API_ENDPOINTS.GET_MACHINES_SOCKET_BASE_URL);
const getUpcomingTrivia = () => client.get(WW_API_ENDPOINTS.GET_UPCOMING_TRIVIA);
const getDigitalGame = (liveArcade = false,isWinnerCircle = false) => client.get(WW_API_ENDPOINTS.GET_DIGITAL_GAME(liveArcade,isWinnerCircle));
const getQRCode = () => client.get(WW_API_ENDPOINTS.GET_QR_CODE);
const getQuickAddTokens = () => client.get(WW_API_ENDPOINTS.GET_QUICK_ADD_TOKENS);
const postTransferNFT = (data) => client.post(WW_API_ENDPOINTS.POST_TRANSFER_NFT, data);
const getDigitalGameGateway = () => client.get(WW_API_ENDPOINTS.GET_DIGITAL_GAME_GATEWAY);
const putRemovePlayer = () => client.put(WW_API_ENDPOINTS.REMOVE_PLAYER);
const getTriviaWinners = (triviaId) => client.get(WW_API_ENDPOINTS.GET_TRIVIA_WINNERS(triviaId));
const getTriviaById = (triviaId) => client.get(WW_API_ENDPOINTS.GET_TRIVIA_BY_ID(triviaId));







const api = {
  playerChallengeRank,
  getMachines,
  getMachinesPrizes,
  getMachinesSocketBaseUrl,
  getDigitalGame,
  getQRCode,
  getQuickAddTokens,
  postTransferNFT,
  getDigitalGameGateway,
  putRemovePlayer,
  getUpcomingTrivia,
  getTriviaWinners,
  getTriviaById
};

export default api;
