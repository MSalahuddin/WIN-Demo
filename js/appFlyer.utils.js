import { Platform } from 'react-native';
import appsFlyer from 'react-native-appsflyer';
import Config from 'react-native-config';
import { requestTrackingPermission } from 'react-native-tracking-transparency';

const config = {
    devKey: Config.APPSFLYER_DEV_KEY,
    isDebug: false,
    appId: Config.APP_STORE_ID,
    onInstallConversionDataListener: true,
    onDeepLinkListener: true,
    timeToWaitForATTUserAuthorization: 60
};

export const AFInit = async () => {
    try {
        await appsFlyer.initSdk(config);
        await requestTrackingPermission()
        return Promise.resolve()
    } catch (error) {
        return Promise.resolve()
    }
}

export function AFLogCustomEvent(name, values) {
    appsFlyer.logEvent(name, values, (res) => {
    }, (error) => {
    });
}

export function AFsetUserIdentityEvent(userId) {
    appsFlyer.setCustomerUserId(userId, (res) => {
    },(error) => {
    });
}

export const AFLogStandardEvent = async (eventName, eventProperties = null) => {
    appsFlyer.logEvent(eventName, eventProperties, (res) => {
    }, (error) => {
    });
}
