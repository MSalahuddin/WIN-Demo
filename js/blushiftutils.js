import { NativeModules } from 'react-native'

const { BlueshiftBridge } = NativeModules;


export const logBlushiftEvent = (eventName, eventProperties = null) => BlueshiftBridge.trackCustomEvent(eventName, eventProperties, false)
export const setBlueShiftCustomerInfo = (playerId) => BlueshiftBridge.setUserInfoCustomerId(playerId)
export const setBlueShiftCustomerEmailId = (email) => BlueshiftBridge.setUserInfoEmailId(email)
