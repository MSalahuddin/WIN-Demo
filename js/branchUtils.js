import branch, { BranchEvent } from 'react-native-branch';

export const setUserIdentity = async (userId) => {
    branch.setIdentity(userId);
}
export const logBranchCustomEvent = async (eventName, eventProperties = null) => {
    new BranchEvent(eventName, null, eventProperties).logEvent();
}   
export const logBranchStandardEvent = async (eventName, eventProperties = null) => {
  const event =  new BranchEvent(eventName, null, eventProperties);
  event.logEvent();
 }