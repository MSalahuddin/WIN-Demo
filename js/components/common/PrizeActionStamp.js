import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

import Text, { FONT_FAMILY, SIZE } from './Text';
import { scale, scaleHeight } from '../../platformUtils';
import { parseDatetime } from '../../utils';
import { color } from '../../styles';
import { PRIZE_STATUS } from '../../constants';

const OuterCircle = styled.View`
  width: ${scale(106)};
  height: ${scale(106)};
  border-radius: ${scale(53)};
  border-width: ${scale(3)};
  border-color: ${({ bodyColor }) => bodyColor};
  justify-content: center;
  align-items: center;
  overflow: hidden;
`;

const InnerCircle = styled.View`
  width: ${scale(95)};
  height: ${scale(95)};
  border-width: ${scale(1)};
  border-color: ${({ bodyColor }) => bodyColor};
  border-radius: ${scale(47.5)};
  justify-content: center;
  align-items: center;
  transform: rotate(-30deg);
`;

const TextStrip = styled.View`
  height: ${scaleHeight(29)};
  width: ${scale(101)};
  background-color: ${({ bodyColor }) => bodyColor};
  justify-content: center;
  align-items: center;
  margin-bottom: ${scaleHeight(3)};
`;

const CenterText = styled(Text)`
  color: ${color.white};
  margin-top: ${scaleHeight(6)};
`;

const DatetimeText = styled(Text)`
  color: ${({ bodyColor }) => bodyColor};
`;

const ContentBody = styled.View`
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const PrizeActionStamp = ({ datetime, type }) => {
  const getStampColor = () => {
    switch (type) {
      case PRIZE_STATUS.SWAPPED:
        return color.blue;
      case PRIZE_STATUS.SHIPPED:
        return color.green;
      case PRIZE_STATUS.CLAIMED:
        return color.green;
      default:
        return color.pinkRed;
    }
  };
  const stampColor = getStampColor();
  const { date, time } = parseDatetime(datetime);
  return (
    <OuterCircle bodyColor={stampColor}>
      <InnerCircle bodyColor={stampColor}>
        <ContentBody>
          <DatetimeText bodyColor={stampColor} size={SIZE.XXXSMALL}>
            {date}
          </DatetimeText>
          <TextStrip bodyColor={stampColor}>
            <CenterText fontFamily={FONT_FAMILY.BOLD} size={SIZE.XSMALL}>
              {type}
            </CenterText>
          </TextStrip>
          <DatetimeText bodyColor={stampColor} size={SIZE.XXXSMALL}>
            {time}
          </DatetimeText>
        </ContentBody>
      </InnerCircle>
    </OuterCircle>
  );
};

PrizeActionStamp.propTypes = {
  datetime: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  type: PropTypes.oneOf([
    PRIZE_STATUS.SWAPPED,
    PRIZE_STATUS.SHIPPED,
    PRIZE_STATUS.PENDING,
    PRIZE_STATUS.NEW,
    PRIZE_STATUS.PACKING
  ]).isRequired
};

export default PrizeActionStamp;
