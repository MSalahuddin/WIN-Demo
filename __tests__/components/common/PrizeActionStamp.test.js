import React from 'react';
import { render } from 'react-native-testing-library';

import PrizeActionStamp from '../../../js/components/common/PrizeActionStamp';
import { PRIZE_STATUS } from '../../../js/constants';

describe('Component PrizeActionStamp', () => {
  const timestamp = '2012-01-26T13:51:50.417-07:00';
  test('renders correctly - prize not been swapped or shipped.', () => {
    const output = render(<PrizeActionStamp type={PRIZE_STATUS.NEW} datetime={timestamp} />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly when type="shipped"', () => {
    const output = render(<PrizeActionStamp type={PRIZE_STATUS.SHIPPED} datetime={timestamp} />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly when type="swapped"', () => {
    const output = render(<PrizeActionStamp type={PRIZE_STATUS.SWAPPED} datetime={timestamp} />);
    expect(output).toMatchSnapshot();
  });
  test('renders correctly when type="packing"', () => {
    const output = render(<PrizeActionStamp type={PRIZE_STATUS.PACKING} datetime={timestamp} />);
    expect(output).toMatchSnapshot();
  });
});
