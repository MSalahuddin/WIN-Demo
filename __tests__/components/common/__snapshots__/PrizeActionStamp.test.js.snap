// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrizeActionStamp renders correctly - prize not been swapped or shipped. 1`] = `
<View
  bodyColor="#F06B59"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#F06B59",
        "borderRadius": 79.5,
        "borderWidth": 4.5,
        "height": 159,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 159,
      },
    ]
  }
>
  <View
    bodyColor="#F06B59"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F06B59",
          "borderRadius": 71.25,
          "borderWidth": 1.5,
          "height": 142.5,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotate": "-30deg",
            },
          ],
          "width": 142.5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#F06B59"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#F06B59",
            },
          ]
        }
      >
        Jan 27, 2012
      </Text>
      <View
        bodyColor="#F06B59"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F06B59",
              "height": 58,
              "justifyContent": "center",
              "marginBottom": 6,
              "width": 151.5,
            },
          ]
        }
      >
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#000000"
          fontFamily="BOLD"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 24,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "color": "#FFFFFF",
                "marginTop": 12,
              },
            ]
          }
        >
          New
        </Text>
      </View>
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#F06B59"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#F06B59",
            },
          ]
        }
      >
        01:51 AM
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component PrizeActionStamp renders correctly when type="packing" 1`] = `
<View
  bodyColor="#F06B59"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#F06B59",
        "borderRadius": 79.5,
        "borderWidth": 4.5,
        "height": 159,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 159,
      },
    ]
  }
>
  <View
    bodyColor="#F06B59"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#F06B59",
          "borderRadius": 71.25,
          "borderWidth": 1.5,
          "height": 142.5,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotate": "-30deg",
            },
          ],
          "width": 142.5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#F06B59"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#F06B59",
            },
          ]
        }
      >
        Jan 27, 2012
      </Text>
      <View
        bodyColor="#F06B59"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F06B59",
              "height": 58,
              "justifyContent": "center",
              "marginBottom": 6,
              "width": 151.5,
            },
          ]
        }
      >
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#000000"
          fontFamily="BOLD"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 24,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "color": "#FFFFFF",
                "marginTop": 12,
              },
            ]
          }
        >
          Packing
        </Text>
      </View>
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#F06B59"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#F06B59",
            },
          ]
        }
      >
        01:51 AM
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component PrizeActionStamp renders correctly when type="shipped" 1`] = `
<View
  bodyColor="#5EC98C"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#5EC98C",
        "borderRadius": 79.5,
        "borderWidth": 4.5,
        "height": 159,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 159,
      },
    ]
  }
>
  <View
    bodyColor="#5EC98C"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#5EC98C",
          "borderRadius": 71.25,
          "borderWidth": 1.5,
          "height": 142.5,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotate": "-30deg",
            },
          ],
          "width": 142.5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#5EC98C"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#5EC98C",
            },
          ]
        }
      >
        Jan 27, 2012
      </Text>
      <View
        bodyColor="#5EC98C"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#5EC98C",
              "height": 58,
              "justifyContent": "center",
              "marginBottom": 6,
              "width": 151.5,
            },
          ]
        }
      >
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#000000"
          fontFamily="BOLD"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 24,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "color": "#FFFFFF",
                "marginTop": 12,
              },
            ]
          }
        >
          Shipped
        </Text>
      </View>
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#5EC98C"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#5EC98C",
            },
          ]
        }
      >
        01:51 AM
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component PrizeActionStamp renders correctly when type="swapped" 1`] = `
<View
  bodyColor="#7C89D8"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#7C89D8",
        "borderRadius": 79.5,
        "borderWidth": 4.5,
        "height": 159,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 159,
      },
    ]
  }
>
  <View
    bodyColor="#7C89D8"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#7C89D8",
          "borderRadius": 71.25,
          "borderWidth": 1.5,
          "height": 142.5,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotate": "-30deg",
            },
          ],
          "width": 142.5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#7C89D8"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#7C89D8",
            },
          ]
        }
      >
        Jan 27, 2012
      </Text>
      <View
        bodyColor="#7C89D8"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#7C89D8",
              "height": 58,
              "justifyContent": "center",
              "marginBottom": 6,
              "width": 151.5,
            },
          ]
        }
      >
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#000000"
          fontFamily="BOLD"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 24,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "color": "#FFFFFF",
                "marginTop": 12,
              },
            ]
          }
        >
          Swapped
        </Text>
      </View>
      <Text
        alignCenter={false}
        allowFontScaling={true}
        bodyColor="#7C89D8"
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 19.5,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#7C89D8",
            },
          ]
        }
      >
        01:51 AM
      </Text>
    </View>
  </View>
</View>
`;
